packages/components/src/FeatureSwitch/FeatureSwitch.tsx

```diff
@@ -78,13 +78,18 @@ export function FeatureSwitch({
   hasSaveIndicator = false,
   title,
 }: FeatureSwitchProps) {
+  const [previousEnabled, setPreviousEnabled] = useState(enabled);
   const [savedIndicator, setSavedIndicator] = useState(false);
   const featureContentClassnames = classnames(
     styles.featureContent,
     styles.content,
     enabled && styles.enabled,
   );
-  shouldShowSavedIndicator();
+
+  if (previousEnabled !== enabled) {
```

Comment:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes